src/cockpit/beiboot.py

```diff
@@ -185,9 +191,21 @@ async def do_connect_transport(self) -> None:
         cmd: Sequence[str] = ('python3', '-ic', '# cockpit-bridge')
         env: Sequence[str] = ()
 
+        in_flatpak = os.path.exists('/.flatpak-info')
+
         # Remote host?  Wrap command with SSH
         if self.destination != 'localhost':
-            ssh_askpass = ensure_ferny_askpass()
+            if in_flatpak:
+                # we run ssh and thus the helper on the host, always use the xdg-cache helper
+                ssh_askpass = ensure_ferny_askpass()
+            else:
+                # outside of the flatpak we expect cockpit-ws and thus an installed helper
+                askpass = patch_libexecdir('${libexecdir}/cockpit-askpass')
+                assert isinstance(askpass, str)
```

Comment:
This assert shouldn't be necessary.

Take a look at the definition of the `J` typevar:


```python
    J = TypeVar('J', JsonObject, JsonDocument)
```

I feel like if you fully-expanded JsonDocument you'd get the correct result.  So like

```
   J = TypeVar('J', str, float, JsonObject, JsonList, ...etc...)
```

...but probably *not* `JsonDocument`.

and if that works then probably remove the comment above it above it being a HACK.  I have a bit of a better understanding of what's going on here than I did when I first wrote this.

Reply:
If the above fails, then just adding `str` to the front of the list ought to suffice, but then leave the comment (or adjust it). :)

Reply:
This seems to be a limitation in mypy.  Let's leave well enough alone for now.